meta-resin-common/recipes-connectivity/redsocks/files/0003-Add-OS-dependent-default-configuration-values.patch

```diff
@@ -44,26 +44,34 @@ index c4bd066..5b5f9e4 100644
 +	instance->config.bindaddr.sin_addr.s_addr = inet_addr("10.114.103.1");
 +	instance->config.bindaddr.sin_port = 5313;
  	instance->config.relayaddr.sin_family = AF_INET;
- 	instance->config.relayaddr.sin_port = htons(53);
+-	instance->config.relayaddr.sin_port = htons(53);
++	instance->config.relayaddr.sin_port = 53;
  	instance->config.relay_timeout = 30;
```

Comment:
is that htons removal necessary?

Reply:
I had to think what this was about. IIt is a bug in upstream - I will upstream too. The default value is set here: https://github.com/darkk/redsocks/blob/5df6a30e5f8e75455e851f2ef87c647eb3b08dff/dnsu2t.c#L342. And that gets set at the start of configuration parsing. Later on, in the parsing finish function, it converts it with htons (https://github.com/darkk/redsocks/blob/5df6a30e5f8e75455e851f2ef87c647eb3b08dff/dnsu2t.c#L368). So everything works fine if the user sets a sin_port. If the default value is used, it will be converted with htons twice which obviously breaks on little endian.

Reply:
I see